addons/mail/views/mail_activity_views.xml

```diff
@@ -123,7 +123,7 @@
                         </group>
                         <group>
                             <field name="date_deadline"/>
-                            <field name="user_id"/>
+                            <field name="user_id" domain="[('groups_id', 'in', %(base.group_user)d)]"/>
```

Comment:
Actually there is an even simpler solution, using the ``share`` field on user :)

Reply:
Done!